packages/react/src/components/ChoiceGroup/ChoiceGroup.base.tsx

```diff
@@ -41,9 +40,8 @@ const focusSelectedOption = (options: IChoiceGroupOption[], keyChecked: string |
   }
 };
 
-// Test if focus came from a sibling DOM element
-const focusFromSibling = (evt: React.FocusEvent<HTMLElement>): boolean => {
-  return !!(evt.relatedTarget && !elementContains(evt.currentTarget, evt.relatedTarget as HTMLElement));
+const focusFromFocusTrapZone = (evt: React.FocusEvent<HTMLElement>): boolean => {
+  return !!(evt.relatedTarget && (evt.relatedTarget as HTMLElement)?.dataset.isFocusTrapZoneBumper === 'true');
```

Comment:
Is it safe to assume that `evt.relatedTarget` is always an HTMLElement? Maybe add an `instanceof` check instead?

```suggestion
  return !!((evt.relatedTarget instanceof HTMLElement) && evt.relatedTarget.dataset.isFocusTrapZoneBumper === 'true');
```